gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Host.java

```diff
@@ -49,7 +49,8 @@ public final class Host {
     Host(final InetSocketAddress address, final Cluster cluster) {
         this.cluster = cluster;
         this.address = address;
-        this.hostUri = makeUriFromAddress(address, cluster.getPath(), cluster.connectionPoolSettings().enableSsl);
+        this.hostUri = makeUriFromAddress(address, cluster.getPath(), cluster.connectionPoolSettings().enableSsl,
```

Comment:
Suggestion: It'd be great if the driver can allow a user to send a request with custom path and custom request header.

It may give too much freedom but at the same time, when graph providers offer additional http endpoints an user can still use the driver to call such endpoints. For example, Neptune provides its own explain / profile endpoint 
https://docs.aws.amazon.com/neptune/latest/userguide/gremlin-profile-api.html

It'd be great if we can support it at driver level so that users don't need to switch clients.

Reply:
yes, i often dislike that the Java driver works the way that it does, at least insofar as how it configures host/port/scheme. like you have to do:

```groovy
g = traversal().withRemote(
                DriverRemoteConnection.using("localhost", 8182))
```

but would be nice if you could do like the other drivers:

```groovy
g = traversal().withRemote(
                new DriverRemoteConnection('ws://localhost:8182/gremlin'))
```

to provide a bit more flexibility for providers and how they setup their servers. that said, imo, i do think Neptune has gone a little too far in creating its own explain/profile endpoints. it should simply implement the `explain()` / `profile()` steps.  For what it's worth, you can use the interceptor interface to add a custom request header now.

Anyway, feel free to open a JIRA for allowing users to specify a url to a "gremlin endpoint" rather than our constructing one. And if you feel like we need to somehow support other custom endpoints, I think you should start a DISCUSS thread on the dev list.

Reply:
Got it, yeah for explain and profile, supporting that on Neptune side and the driver should just add `explain()` or `profile()` in a query, I agree.
It's too much to support more other endpoints like /status, bulkload for Neptune so I don't think at this point adding a custom endpoint is critical at least for Neptune.